ObstacleMSG[] previous_obstacles
TrajectoryMSG robot_trajectory
LeftHMSG[] h_signatures
---
int32 chosen_guidance